src/persist/src/nemesis/generator.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use rand::prelude::SmallRng;
+use rand::{Rng, SeedableRng};
+
+use crate::nemesis::{
+    Input, ReadSnapshotReq, Req, ReqId, SealReq, SnapshotId, TakeSnapshotReq, WriteReq,
+};
+
+pub struct Generator {
+    seed: u64,
+    req_id: u64,
+    snap_id: u64,
+}
+
+// TODO: Make Generator configurable, so individual tests can be written to
+// stress interesting combinations of random traffic.
+impl Generator {
+    pub fn from_seed(seed: u64) -> Self {
+        Generator {
+            seed,
+            req_id: 0,
+            snap_id: 0,
+        }
+    }
+
+    pub fn gen(&mut self) -> Input {
+        let req_id = ReqId(self.req_id);
+        self.req_id += 1;
+        let mut rng = SmallRng::seed_from_u64(self.seed + req_id.0);
+        let req = loop {
+            break match rng.gen_range(0..=3) {
+                0 => {
+                    let stream = rng.gen_range(0..5).to_string();
+                    let key = rng.gen_range('a'..'e').to_string();
```

Comment:
Is this thing going to be fired against actual S3 or a S3-compatible local service? If so, some larger keys may benefit so that the PUT and GET operations start having more varied execution times in order to shake out any race conditions.